### PR TITLE
Only show partners without service areas on map

### LINKED ISSUE #352: [Bug]: All partners map is being stretched out to show Exeter
## Acceptance Criteria

- [ ] If a partner has a service area, they do not have a pin in the map.

Note: The rationale for this is that only venues where people can go should be pinned, not head offices. The Place Cal data is structured such that if an organisation services an area, they list themselves with a service area. If an organisation is a venue that people can go to, they do not list themselves as a service area.

## Description

The map is being stretched out to Exeter as the LGBT consortium have a postal address there. I think it should just focus on London.

Screenshot 2022-10-27 at 15 04 37

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Fixes #352

On /partners only show pins on the map from partners with no service areas

### CODE REVIEW

**Comment by erbridge** [comment]
Isn't this backwards? From my reading, we should only be showing partners _with_ service areas, right?

**Review by erbridge — APPROVED** [review]
Looks sensible. I'd prefer it if we pulled this logic into a self-documenting function to explain what it's for, eg `partnerIsVenueWithLocation` or something, so we have a better chance of working out why we're doing this again later.

File: src/Page/Partners.elm
Comment (erbridge) [inline src/Page/Partners.elm:121]:
```suggestion
        partnersIsVenueWithLocation partner =
            List.isEmpty partner.areasServed && (partner.maybeGeo /= Nothing)
```

File: src/Page/Partners.elm
Comment (erbridge) [inline src/Page/Partners.elm:124]:
```suggestion
        partnerToGeo partner =
```

Comment (ivan-kocienski-gfsc) [reply src/Page/Partners.elm:124]: This is how the elm formatter likes it

File: src/Page/Partners.elm
Comment (erbridge) [reply src/Page/Partners.elm:121]: This suggestion has two changes:

1. Renaming the function
2. Changing the function signature

Even if we disagree on (1), (2) is worth considering. It's _not_ a formatting change.

File: src/Page/Partners.elm
Comment (ivan-kocienski-gfsc) [reply src/Page/Partners.elm:124]: ah i see it now. we want it to be a lambda that takes one argument so List.filter/map can use it

Comment (erbridge) [reply src/Page/Partners.elm:124]: I think you misunderstand how this works in Elm. It's perfectly legal (and normal) to pass functions around, not lambdas. This lambda assignment looks odd and isn't something we do elsewhere in this codebase.

Comment (erbridge) [reply src/Page/Partners.elm:124]: The point of the lambda / anonymous function is to be able to define a function inline, which we're no longer doing by refactoring in this way.

Comment (ivan-kocienski-gfsc) [reply src/Page/Partners.elm:124]: Perhaps I am committing a faux pa by using a lisp-ism but it is okay to define a lambda and assign it to a variable and then pass that variable in to a HLF like map. 

I changed it and it works so I guess you can define helper methods multiple ways.

**Comment by katjam** [comment]
@ivan-kocienski-gfsc I'd like to merge this, if you are happy.

Usually I'd wait for the author to merge, but since we are launching on Tues, I want to get these in asap. I'll leave it til Saturday morning and if I've not heard, i'll go ahead.

**Pull request merged**
